pkg/integrations/quay.go

```diff
@@ -0,0 +1,182 @@
+	p.StringVar(&q.dockerconfigjson, "dockerconfigjson", q.dockerconfigjson,
+		"Quay dockerconfigjson")
+	p.StringVar(&q.token, "token", q.token,
+		"Quay API token")
+	p.StringVar(&q.url, "url", q.url,
+		"Quay URL")
+}
+
+// log logger with contextual information.
+func (q *QuayIntegration) log() *slog.Logger {
+	return q.logger.With(
+		"url", q.url,
+		"force", q.force,
+		"dockerconfigjson-len", len(q.dockerconfigjson),
+		"token-len", len(q.token),
+	)
+}
+
+// Validate checks if the required configuration is set.
+func (q *QuayIntegration) Validate() error {
+	if q.dockerconfigjson == "" {
+		return fmt.Errorf("dockerconfigjson is required")
+	}
+	if q.token == "" {
+		return fmt.Errorf("token is required")
+	}
+	if q.url == "" {
+		return fmt.Errorf("url is required")
+	} else {
+		u, err := url.Parse(q.url)
+		if err != nil {
+			return fmt.Errorf("invalid url")
+		}
+		if !strings.HasPrefix(u.Scheme, "http") {
+			return fmt.Errorf("invalid url scheme, expected one of 'http', 'https'")
+		}
+	}
+	return nil
+}
+
+// EnsureNamespace ensures the namespace needed for the Quay integration secret
+// is created on the cluster.
+func (q *QuayIntegration) EnsureNamespace(ctx context.Context) error {
+	feature, err := q.cfg.GetFeature(config.RedHatDeveloperHub)
+	if err != nil {
+		return err
+	}
+	return k8s.EnsureOpenShiftProject(
+		ctx,
+		q.log(),
+		q.kube,
+		feature.GetNamespace(),
+	)
+}
+
+// secretName returns the secret name for the integration. The name is "lazy"
+// generated to make sure configuration is already loaded.
+func (q *QuayIntegration) secretName() types.NamespacedName {
+	feature, _ := q.cfg.GetFeature(config.RedHatDeveloperHub)
+	return types.NamespacedName{
+		Namespace: feature.GetNamespace(),
+		Name:      "rhtap-quay-integration",
+	}
+}
+
+// prepareSecret checks if the secret already exists, and if so, it will delete
+// the secret if the force flag is enabled.
+func (q *QuayIntegration) prepareSecret(ctx context.Context) error {
+	q.log().Debug("Checking if integration secret exists")
+	exists, err := k8s.SecretExists(ctx, q.kube, q.secretName())
+	if err != nil {
+		return err
+	}
+	if !exists {
+		q.log().Debug("Integration secret does not exist")
+		return nil
+	}
+	if !q.force {
+		q.log().Debug("Integration secret already exists")
+		return fmt.Errorf("%w: %s",
+			ErrSecretAlreadyExists, q.secretName().String())
+	}
+	q.log().Debug("Integration secret already exists, recreating it")
+	return k8s.DeleteSecret(ctx, q.kube, q.secretName())
+}
+
+// store creates the secret with the integration data.
+func (q *QuayIntegration) store(
+	ctx context.Context,
+) error {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: q.secretName().Namespace,
+			Name:      q.secretName().Name,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"dockerconfigjson": []byte(q.dockerconfigjson),
+			"token":            []byte(q.token),
+			"url":              []byte(q.url),
+		},
+	}
```

Comment:
We should follow common Kubernetes convention to have a secret typed `docker-registry`, the only adaptation needed here is storing the `--from-file` argument as `.dockerconfigjson` attribute. 🙏

Reply:
Updated to use the requested secret type. The other comment will track the use of `--from-file`.